FactoryTest/FactoryTest.vcxproj

```diff
@@ -73,7 +73,7 @@
       <FunctionLevelLinking>true</FunctionLevelLinking>
       <IntrinsicFunctions>true</IntrinsicFunctions>
       <SDLCheck>true</SDLCheck>
-      <AdditionalIncludeDirectories>$(ProjectDir)..\FactoryTest;$(ProjectDir)..\..\..\Arduino\libraries\ButtonDebounce\src;$(ProjectDir)..\..\..\Arduino\libraries\FanController;$(ProjectDir)..\..\..\..\..\..\Program Files (x86)\Arduino\hardware\arduino\avr\cores\arduino;$(ProjectDir)..\..\..\..\..\..\Program Files (x86)\Arduino\hardware\arduino\avr\variants\eightanaloginputs;$(ProjectDir)..\..\..\..\DOCUME~1\code\PAPR-F~1\FACTOR~1;$(ProjectDir)..\..\..\..\..\..\Program Files (x86)\Arduino\hardware\tools\avr\\lib\gcc\avr\7.3.0\include;$(ProjectDir)..\..\..\..\..\..\Program Files (x86)\Arduino\hardware\tools\avr\avr\include;$(ProjectDir)..\..\..\..\..\..\Program Files (x86)\Arduino\hardware\tools\avr\\lib\gcc\avr\7.3.0\include;$(ProjectDir)..\..\..\..\..\..\Program Files (x86)\Arduino\hardware\tools\avr\avr\include-fixed;$(ProjectDir)..\..\..\..\..\..\Program Files (x86)\Arduino\hardware\tools\avr\avr\include\avr;$(ProjectDir)..\..\..\..\..\..\Program Files (x86)\Arduino\hardware\tools\avr\lib\gcc\avr\4.9.2\include;$(ProjectDir)..\..\..\..\..\..\Program Files (x86)\Arduino\hardware\tools\avr\lib\gcc\avr\4.9.2\include;$(ProjectDir)..\..\..\..\..\..\Program Files (x86)\Arduino\hardware\tools\avr\lib\gcc\avr\4.9.3\include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <AdditionalIncludeDirectories>$(ProjectDir)..\FactoryTest;$(ProjectDir)..\..\..\..\..\..\Program Files (x86)\Arduino\hardware\arduino\avr\cores\arduino;$(ProjectDir)..\..\..\..\..\..\Program Files (x86)\Arduino\hardware\arduino\avr\variants\eightanaloginputs;$(ProjectDir)..\..\..\..\DOCUME~1\code\PAPR-F~1\FACTOR~1;$(ProjectDir)..\..\..\..\..\..\Program Files (x86)\Arduino\hardware\tools\avr\\lib\gcc\avr\7.3.0\include;$(ProjectDir)..\..\..\..\..\..\Program Files (x86)\Arduino\hardware\tools\avr\avr\include;$(ProjectDir)..\..\..\..\..\..\Program Files (x86)\Arduino\hardware\tools\avr\\lib\gcc\avr\7.3.0\include;$(ProjectDir)..\..\..\..\..\..\Program Files (x86)\Arduino\hardware\tools\avr\avr\include-fixed;$(ProjectDir)..\..\..\..\..\..\Program Files (x86)\Arduino\hardware\tools\avr\avr\include\avr;$(ProjectDir)..\..\..\..\..\..\Program Files (x86)\Arduino\hardware\tools\avr\lib\gcc\avr\4.9.2\include;$(ProjectDir)..\..\..\..\..\..\Program Files (x86)\Arduino\hardware\tools\avr\lib\gcc\avr\4.9.2\include;$(ProjectDir)..\..\..\..\..\..\Program Files (x86)\Arduino\hardware\tools\avr\lib\gcc\avr\4.9.3\include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
```

Comment:
Program files may be in a different location on people's machines relative to your home directory. I believe there is an envvar for this.

Reply:
same comment about Documents and settings

Reply:
I investigated and found that every time you do a build, visual studio uses the current setting of `Extensions > vMicro > IDE > Configure IDE Location` for its include path, and then writes those absolute paths to the vcproj file, overwriting whatever was there before. In other words, it ignores what's in the vcproj file, so I changed the vcproj file to have a placeholder string. In future we can try to always strip out the absolute paths before committing a new version of the vcxproj.

Reply:
I looked around and couldn't find an environment variable for the Arduino IDE path, nor for the Arduino sketchbook path. 
There is however a Visual Micro setting `Extensions > vMicro > General > Configure Arduino IDE Location`.

I investigated and found that every time you do a build, visual micro uses the current setting of `Configure Arduino IDE Location` for its include path, and then writes those absolute paths to the vcproj file, overwriting whatever was there before. In fact, it seems to completely ignores what's in the vcproj file, so I simply changed the vcproj file to have a placeholder string. In the future, if we need to modify the vcxproj file, we'll have to remember to manually remove the absolute paths before we commit it. (Although if we don't, it may not actually cause any trouble).
 
 I have no idea where Visual Micro gets the Arduino Sketchbook path from, and why it writes the path to the vcxproj file.
 I replaced the path with a placeholder and that doesn't seem to cause any problems. The sketchbook location inside the
 Documents folder is a default set by the Arduino installer, they obviously don't know that all the cool kids use %HOME% (and apparently, neither do I!)

Reply:
https://www.google.com/search?client=firefox-b-1-d&q=program+files+environment+variable

or tell them to set it... 

ok that works I guess.